### PR TITLE
fix: default asset behaviour for trade input

### LINKED ISSUE #4274: Trade modal not refreshing when needed
### Overview

When entering details for a trade, the trade modal keeps the amounts even when you switch pages, complete a trade or switch wallets on a completed trade screen

### References and additional details

1) Log into app.shapeshift and connect to wallet of choice
2) Navigate to trade page and complete a trade
3) Note the trade's completed tick then select dashboard
4) Note the trade modal has the details of the last completed trade
image
image

**note the trade modal did not refresh when switching wallets on a completed trade screen
**when refreshing native wallet after completed a trade out of eth on optimism, the trade modal default pair became eth on optimism and optimism
image

### Acceptance Criteria

Trade modal should refresh after completing a trade or switching paged or wallets

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

_No response_

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### LINKED ISSUE #4389: Trade pair can be ETH -> ETH
### Overview

Its currently possible to get the app into a state where the trade modal displays ETH and ETH which is invalid.

Reproducible consistently if you go to an asset page from the dashboard.

### References and additional details

image

### Acceptance Criteria

Should not be possible to have the same asset on sell and buy in the trade modal.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

2 hours

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

This fixes how the app automatically selects assets for trade input depending on the context. The following scenarios are now handled:

- Asset page - the asset will automatically be set the the buy asset in the trade input
- Fox opportunities page - the relevant fox asset will automatically be set the the buy asset in the trade input
- Arkeo page - when clicking "trade for fox" the app will now set the buy asset to fox before switching to the trade page

It was also possible to create a state where the buy and sell assets were the same (see linked issue below). This is now fixed

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #4274 (see latest comments)
closes #4389

## Risk

If these fixes are not implemented correctly, there are risks of the following:

- assets being set to the wrong token in the trade input when using the app
- the same asset could be set to buy and sell in the trade input

## Testing

- tested all known uses of trade input and that the same asset cannot be selected for buy and sell
- tested that the correct asset is set to the buy asset for the following
  - fox opportunities page
  - asset page
  - arkeo page

### Engineering

The fix for duplicate assets was to not use `updateBuyAsset` and instead use `handleAssetSelection` which includes logic to dedupe buy+sell assets. There are no other usages of `updateBuyAsset`, so I suggest we remove it to prevent potential regressions in the future, along with `updateSellAsset` which also isnt used anywhere.

### Operations

This closes 2 known production issues which are user facing. Please check the linked issues to confirm.

### CODE REVIEW

File: src/components/AssetAccountDetails.tsx [hunk 1]
```diff
@@ -30,18 +29,6 @@ export const AssetAccountDetails = ({ assetId, accountId }: AssetDetailsProps) =
...
-  const clearAmounts = useSwapperStore(state => state.clearAmounts)
-  const updateBuyAsset = useSwapperStore(state => state.updateBuyAsset)
-
-  const feeAsset = useAppSelector(state => selectFeeAssetById(state, assetId))
-  const buyAsset = useAppSelector(state => selectAssetById(state, assetId))
-
-  useEffect(() => {
-    const buyAssetWithFallback = buyAsset ?? feeAsset
-    if (buyAssetWithFallback) updateBuyAsset(buyAssetWithFallback)
-    clearAmounts()
```
Comment (0xApotheosis) [inline src/components/AssetAccountDetails.tsx:42]: By removing this I was able to observe inconsistent behaviour, whereby now we sometimes clear the amounts when changing routes, and sometimes we don't.

We might need to consider the product implications of not clearing amounts between pages, as we don't want stale quotes floating around.

We did have a quote retry mechanism in place, though I'm not sure it is still working.

File: src/components/AssetAccountDetails.tsx [hunk 2]
```diff
@@ -30,18 +29,6 @@ export const AssetAccountDetails = ({ assetId, accountId }: AssetDetailsProps) =
   const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
   const assetIds = useMemo(() => [assetId], [assetId])
 
-  const clearAmounts = useSwapperStore(state => state.clearAmounts)
-  const updateBuyAsset = useSwapperStore(state => state.updateBuyAsset)
```
Comment (0xApotheosis) [inline src/components/AssetAccountDetails.tsx:34]: As per your comment, lets rip out `updateBuyAsset` and `updateSellAsset` - great catch.

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
This does what it is intended to do, though introduces one regression with the amounts clearing (see [comment](https://github.com/shapeshift/web/pull/4390#discussion_r1182003260)).

Let's also rip out the now unused actions 🤝

File: src/components/AssetAccountDetails.tsx (hunk 1 above)
Comment (woodenfurniture) [reply src/components/AssetAccountDetails.tsx:42]: yeah we are actually doing this in prod where quotes values are left as-in in the trade input state until something is changed. Only hitting a page where a default is provided will clear the state, which is the logic here too. But I agree that to the user this appears inconsistent and we dont have solid logic to refresh quotes. 

I'm going to make the trade input clear the values on every load for consistency.

Comment (0xApotheosis) [reply src/components/AssetAccountDetails.tsx:42]: Yeah right. If it's in prod we should for sure get that fixed - double whammy PR 🙏

**Review by 0xApotheosis — APPROVED** [review]
🍬

**Pull request merged**